### PR TITLE
Feature/max rv filter

### PR DESCRIPTION
**Pull request by HassnHamada**
## Describe your changes

I've added a feature to filter by maximum potential RV.
This feature has been implemented in the artifact main pages and involved changes to the following files:

* /apps/frontend/src/app/Components/Artifact/ArtifactFilterDisplay.tsx
* /apps/frontend/src/app/PageArtifact/ArtifactSort.ts
* /libs/gi/db/src/Database/DataEntries/DisplayArtifactEntry.ts

max-rv-filter-480@10

## Issue or discord link

- 

## Testing/validation

I have run `yarn run mini-ci` locally, and all tests have passed successfully.

Screenshot_280

**But I didn't add any new tests.**

## Checklistbefore requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Review by frzyc — CHANGES_REQUESTED** [review]
Hey @HassnHamada , Thanks for your contribution. Can you remove the yarn.lock and yarn cache changes? I am not sure they there is so much file churn here, but you should be able to use the current version of the dependencies.

**Comment by HassnHamada** [comment]
@frzyc I just reverted the changes it should be good now. [Here](https://github.com/HassnHamada/genshin-optimizer/compare/bfafba6372aca0b65df455785e0805956e76296c...97401e2183c94727e0b2c2ebcdcc8a7d4cd444bb) is a diff between the base and the last commit.

File: apps/frontend/src/app/Components/Artifact/RVSlide.tsx [hunk 1]
```diff
@@ -45,28 +58,58 @@ export default function RVSlide({
...
-        onChange={(val) => setLow(clamp(val ?? 0, 0, levelHigh))}
-        sx={{ px: 1, pl: 2, width: 100, borderRadius: '4px 0 0 4px' }}
-        inputProps={{ sx: { textAlign: 'right' } }}
-        startAdornment={'RV: '}
-        disabled={disabled}
-      />
+      <Box
+        sx={{ width: 100, height: 30, display: 'flex', alignItems: 'center' }}
+      >
+        <Button
```
Comment (frzyc) [inline apps/frontend/src/app/Components/Artifact/RVSlide.tsx:64]: This behaviour seems kind of complicated, I wonder if it makes sense to just have both RV filters, and remove this `useMaxRV` complexity?

File: libs/gi/db/src/Database/DataEntries/DisplayArtifactEntry.ts [hunk 2]
```diff
@@ -46,6 +46,8 @@ export type FilterOption = {
   locked: Array<'locked' | 'unlocked'>
   rvLow: number
   rvHigh: number
+  mrvLow: number
```
Comment (frzyc) [inline libs/gi/db/src/Database/DataEntries/DisplayArtifactEntry.ts:49]: I think your implementation is flawed from the data perspective, you should have 3 data: `rvLow, rvHigh, useMaxRv`. Currently, the RV values are not persisted when you toggle between `useMaxRv`, and after page refresh, the value of the toggle is not persisted either.

File: apps/frontend/src/app/Components/Artifact/RVSlide.tsx [hunk 3]
```diff
@@ -45,28 +58,58 @@ export default function RVSlide({
...
+            })
+          }
+          sx={{
+            p: 0,
+            minWidth: 0,
+            width: '50%',
+            height: '100%',
+            borderRadius: '4px 0 0 4px',
+            transition: 'transform 0.3s ease',
+            '&:hover': { transform: 'translate(-3px, -3px)' },
```
Comment (frzyc) [inline apps/frontend/src/app/Components/Artifact/RVSlide.tsx:78]: I am not sure about the additional hover effect on this button, and the fixed width makes this button seemed cramped. My opinion is that you should split the button/input with a <Divider/>, and add more spacing.

**Review by frzyc — DISMISSED** [review]
Some concerns about implementation.

**Review by nguyentvan7 — COMMENTED** [review]
image
padding kinda funky

**Comment by HassnHamada** [comment]
I could remove the padding and center the text. But I was trying to stay consistent with the level slider padding.

Screenshot_281
Screenshot_282

Another idea is to do like [this](https://discord.com/channels/785153694478893126/1209509900460228708/1209621217481334805) comment suggests. It is a good idea to not change the UI, but it may be restrictive for users.

I would like to hear your thoughts.

**Comment by nguyentvan7** [comment]
I like the first option

**Comment by frzyc** [comment]
https://genshin-optimizer-prs.github.io/pr/1517/frontend/

**Comment by frzyc** [comment]
I agree with centering the "RV"/"MRV". I appreciate the consideration you are putting in.

**Comment by HassnHamada** [comment]
This may also be good change. Just to match them.

Screenshot_283
Screenshot_284

PS: Do you prefer communicating on GitHub or Discord?

**Comment by nguyentvan7** [comment]
Oh yea that looks good

Discord is preferred for initial conversation about the feature. once the code is here, it is up to you. If you think there will be a lot of back and forth, discord is probably easier

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**